File: .yalc/@entur/atb-mobile-client-sdk/challenge/bluetooth/model/bluetooth-model-java-lib/src/test/java/no/entur/abt/android/bluetooth/ChallengeAdvertisingFrameFilterTest.java
package no.entur.abt.android.bluetooth;

import static org.junit.jupiter.api.Assertions.fail;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import org.junit.jupiter.api.Test;

public class ChallengeAdvertisingFrameFilterTest {

	private ChallengeAdvertisingFrameFilter filter = new ChallengeAdvertisingFrameFilter();

	@Test
	public void testFilter() {
		Random random = new Random();

		List<ChallengeAdvertisingFrame> frames = new ArrayList<>();
		for (int clientId = 0; clientId < 65535 - 1000; clientId += 1000) {
			for (int correlationId = 0; correlationId < 256 - 10; correlationId += 10) {
				for (int channel = 0; channel < 8 - 2; channel += 2) {
					for (int frameNumber = 0; frameNumber < 16; frameNumber += 1) {
						byte[] content = new byte[23];
						random.nextBytes(content);

						frames.add(
								ChallengeAdvertisingFrameBinaryFormat.newInstance(content, clientId, frameNumber, frameNumber == 15, correlationId, channel));
					}
				}
			}
		}

		for (ChallengeAdvertisingFrame frame : frames) {
			if (filter.hasFrame(frame.toByteArray())) {
				fail();
			}
			if (!filter.hasFrame(frame.toByteArray())) {
				fail();
			}
		}
	}
}

File: .yalc/@entur/atb-mobile-client-sdk/challenge/bluetooth/model/bluetooth-model-ios-lib/Tests/bluetooth-model-ios-libTests/ChallengeAdvertisingFrameFilterTest.swift
//
//  ChallengeFilterTest.swift
//  
//
//  Created by Thomas Skjolberg on 10/06/2022.
//

import XCTest
@testable import bluetooth_model_ios_lib

class ChallengeAdvertisingFrameFilterTest: XCTestCase {

    func random(length: Int) -> Data {
        return Data((0 ..< length).map { _ in UInt8.random(in: UInt8.min ... UInt8.max) })
    }

    var filter = ChallengeAdvertisingFrameFilter()

    func testFilter() throws {
        var frames = Array<Data>()
        frames.reserveCapacity(655 * 25 * 3 * 15)

        for clientId in stride(from: 0, to: (65535-1000), by: 1000) {
            for correlationId in stride(from: 0, to: (256-10), by: 10) {
                for channel in stride(from: 0, to: (8-2), by: 2) {
                    for frameNumber in 0..<16 {
                        var content = random(length: 23)

                        let frame = try ChallengeAdvertisingFrameBinaryFormat.newInstance(payload: &content, offset:0, count:content.count, clientId: UInt32(clientId), frameNumber: UInt32(frameNumber), lastFrame: frameNumber == 15, correlationId: UInt32(correlationId), channel: UInt32(channel))
                        frames.append(frame.contents)
                    }
                }
            }
        }
        
        for var frame in frames {
            XCTAssertFalse(filter.hasFrame(frame: &frame))
            XCTAssertTrue(filter.hasFrame(frame: &frame))
        }
    }
}
